### PR TITLE
:sparkles: Pull Request and Issue Templates

### PR DESCRIPTION
**Pull request by Paulomart**
**Changes:**

1. Add issue template
2. Add pull request template

**Issues:**

PR: #4 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+:tada: **Thanks for submitting this Pull Request**
+
+Please provide a list including your changes.
+
+To help us to create a meaningful Changelog, we agreed
+to use an emoji-based commit-style. Use the [Emojis](#emojis)
+we provided below.
+
+Fill out the following, provide reference to issues
+and this PR- we will use this as merge commit message-body
```
Comment (FreeYourMinds) [inline .github/PULL_REQUEST_TEMPLATE.md]: A dash behind `PR` intended?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+Please check the boxes in this list after submitting your PR:
+
+- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
+- [x] I've tested **all** changes included in this PR.
+- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
+- [x] I've rebased the `develop` branch with my branch before finishing this PR.
+- [x] I've **summarized all changes** in a list above.
+- [x] I've mentioned all **PRs, which relate to this one**.
+- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).
```
Comment (FreeYourMinds) [inline .github/PULL_REQUEST_TEMPLATE.md]: The `is` in the sentence wrong.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+> You can remove this section if it's irrelevant to the issue
+> (for example for certain questions or feature requests).
+
+#### Issue checklist
+
+Please check the boxes in this list after submitting your Issue:
+
+- [ ] I've checked if this issue already exists
+- [ ] I've included all the information that i think is relevant
```
Comment (FreeYourMinds) [inline .github/ISSUE_TEMPLATE.md]: `I` should be uppercase.

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+> 
+> Logs and screenshots are very welcome!
+
+#### How can others reproduce the issue?
+
+> Describe how others can reproduce this issue as precisely as possible.
+> 
+> If you can't reproduce the issue, let us know what you tried!
+> 
+> You can remove this section if the issue isn't something reproducable
```
Comment (FreeYourMinds) [inline .github/ISSUE_TEMPLATE.md]: Should be `reproducible`.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Paulomart) [reply .github/PULL_REQUEST_TEMPLATE.md]: I think so, this is just copied from the template.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
+:tada: **Thanks for submitting this Pull Request**
+
+Please provide a list including your changes.
+
+To help us to create a meaningful Changelog, we agreed
+to use an emoji-based commit-style. Use the [Emojis](#emojis)
+we provided below.
+
+Fill out the following, provide reference to issues
```
Comment (FreeYourMinds) [inline .github/PULL_REQUEST_TEMPLATE.md]: Should be `references`.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (FreeYourMinds) [inline .github/PULL_REQUEST_TEMPLATE.md]: 1. Was a dash behind PR intended?
1. Should be "as **the** merge"

**Comment by Paulomart** [comment]
@cmg-dev I would move the `Template Version` into a comment: 

```markdown
<!-- Template Version 1.1 -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+| New Feature              | `:sparkles:`         | ✨     |
+| Refactoring Code         | `:recycle:`          | ♻️     |
+| Releasing / Version tags | `:bookmark:`         | 🔖     |
+| Removing Stuff           | `:fire:`             | 🔥     |
+| Tests                    | `:white_check_mark:` | ✅     |
+| Work In Progress (WIP)   | `:construction:`     | 🚧     |
+
+</details>
+
+Template Version 1.1
```
Comment (ElRaptorus) [inline .github/PULL_REQUEST_TEMPLATE.md]: I agree with @Paulomart, this should be placed in a comment. Also, might be a good idea to put it at the top of the file.

File: .github/ISSUE_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+#### Issue Checklist
+
+Please check the boxes in this list after submitting your Issue:
+
+- [ ] I've checked if this issue already exists.
+- [ ] I've included all relevant information.
+- [ ] I've added logs and/or screenshots (if applicable).
+- [ ] I've mentioned PRs and issues that relate to this one.
+
+Template Version 1.1
```
Comment (ElRaptorus) [inline .github/ISSUE_TEMPLATE.md]: Same here.

**Comment by Paulomart** [comment]
@ElRaptorus I left it at the bottom because in the pull request template if more likely to get removed when on top.

**Pull request merged**